kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -227,4 +229,16 @@ object Abort:
       */
     inline def catching[E <: Throwable]: CatchingOps[E] = CatchingOps(())
 
+    object internal:
+        final case class Abortable[E] private (classTag: ClassTag[E]) extends AnyVal
+
+        object Abortable:
+            inline given Abortable[Nothing] = Abortable[Nothing](ClassTag.Nothing)
+            inline given [E](using inline ct: ClassTag[E], inline tag: Tag[E]): Abortable[E] =
```

Comment:
It's nice that `tag` is erased by the compiler with this new encoding since it's `inline` and unused.

Reply:
An `inline` parameter gets spliced on each of use. For example:

```scala
inline def test(inline p1: Int): Int = p1 + p1

test(1) // expands to 1 + 1
test(Math.sqrt(42)) // expands to Math.sqrt(42) + Math.sqrt(42)
```

Semantically, an `inline` parameter is equivalent to a by-name parameter in a regular method. If the parameter is never used, it's never spliced and doesn't make it to the generated bytecode.

Reply:
> If it is unused why do we have it?

The `Tag` here is used to disallow handling multiple aborts at once like in `Abort.run[Int | String]`. In those cases, the class tag is inferred as `Any`, which ends up handling all aborts. We might be able to support this scenario eventually but it'd require `Tag` to efficiently encode type unions and provide a way to check tags against the runtime classes of failure values.

Reply:
> Right that should have been obvious. On a bit of a tangent now but I guess that means the compiler still runs the macro but then it gets eliminated in a later phase.

Yep, it's essentially free abstraction :) The tag will only exist at compile time and `Abortable` also won't be in the generated bytecode given that it's an `AnyVal`.

> Ah good point. I wonder if we should alias it or something when it is used for this purpose? Might be clearer to anyone reading it in future. Also if Tag ever changes to support unions it might be tough to find them all.

Makes sense, I'll add a comment clarifying that for now.